ensemble/randomforest.h

```diff
@@ -0,0 +1,53 @@
+#ifndef _MACHINELEARNING_ENSEMBLE_RANDOMFOREST_H_
+#define _MACHINELEARNING_ENSEMBLE_RANDOMFOREST_H_
+
+#include "../tree/tree.h"
+
+namespace ensemble {
+/**
+ * Base class for Random Forest Classification and Regression algorithms.
+ */
+class RandomForest {
+ public:
+  /**
+   * \brief Standard destructor.
+   */
+  virtual ~RandomForest();
+
+  /**
+   * Compute and return the score of the training dataset obtained using an
+   * out-of-bag estimate
+   * \return oob score
+   */
+  virtual double OobScore() = 0;
+
+  /*
+   * Build the RandomForest
+   * \param thread Number of threads for paralel execution
+   * \param tree Number of trees in the Forest
+   */
+  virtual void Build(/* Collection col, */ const int thread,
+                     const int tree) = 0;
+
+  /**
+   * Predicts the labels of the instances of X based on a vote by
+   * the trees of the forest, weighted by their probability estimates
+   * \param Instance labels
```

Comment:
If this function is predicting the class in the context of classification of the regression score for regression problems, why it is required to pass the labels of the samples?
I think that it is not making that much sense. If you are trying to predict something, just pass the collection of samples that will be predicted, and then return a vector containing it's prediction.